pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go

```diff
@@ -0,0 +1,184 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	kube_apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kube_runtime "k8s.io/apimachinery/pkg/runtime"
+	kube_ctrl "sigs.k8s.io/controller-runtime"
+	kube_client "sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	k8s_common "github.com/kumahq/kuma/pkg/plugins/common/k8s"
+	"github.com/kumahq/kuma/pkg/plugins/resources/k8s"
+	k8s_util "github.com/kumahq/kuma/pkg/plugins/runtime/k8s/util"
+	util_k8s "github.com/kumahq/kuma/pkg/util/k8s"
+)
+
+// HTTPRouteReconciler reconciles a GatewayAPI object into Kuma-native objects
+type HTTPRouteReconciler struct {
+	kube_client.Client
+	Reader kube_client.Reader
+	Log    logr.Logger
+
+	Scheme          *kube_runtime.Scheme
+	Converter       k8s_common.Converter
+	SystemNamespace string
+	ResourceManager manager.ResourceManager
+}
+
+// Reconcile handles transforming a gateway-api HTTPRoute into a Kuma
+// GatewayRoute and managing the status of the gateway-api objects.
+func (r *HTTPRouteReconciler) Reconcile(ctx context.Context, req kube_ctrl.Request) (kube_ctrl.Result, error) {
+	httpRoute := &gatewayapi.HTTPRoute{}
+	if err := r.Get(ctx, req.NamespacedName, httpRoute); err != nil {
+		if kube_apierrs.IsNotFound(err) {
+			err = r.ResourceManager.Delete(ctx, core_mesh.NewGatewayRouteResource(), store.DeleteByKey(req.Name, req.Name))
```

Comment:
Should pass mesh as the second parameter to `DeleteByKey`.

Reply:
Removing this call to `ResourceManager.Delete` entirely, it shouldn't be here and doesn't make sense. When the HTTPRoute is deleted the owner ref causes the GatewayRoute to be deleted.